doc/sphinx/source/backends/executable-api/index.rst

```diff
@@ -4,11 +4,29 @@
 Executable
 ==========
 
+* :ref:`generic_executable`
+* :ref:`intelgpu_executable`
+
+
 The ``compile`` function on an ``Executable`` has more direct methods to 
 actions such as ``validate``, ``call``, ``get_performance_data``, and so on. 
 
 
+.. _generic_executable:
+
+Generic executable
+------------------
+
 .. doxygenclass:: ngraph::runtime::Executable
    :project: ngraph
    :members: 
 
+
+.. _intelgpu_executable:
```

Comment:
Should we actually document the specific subclasses here? I kinda don't think so---they are implementing an interface and not instantiated directly (they come from `Backend::compile`), so they should be considered an implementation detail IMHO. (This comment probably also applies to the `Backend` stuff above.)